### PR TITLE
DOC: Improved doc string for IntervalIndex + related changes

### PR DESCRIPTION
**Pull request by topper-123**
- [x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

This PR has Improvements for the doc string for ``IntervalIndex`` + related changes

### CODE REVIEW

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -120,9 +120,42 @@ class IntervalIndex(IntervalMixin, Index):
...
+    ---------
+    A new ``IntervalIndex`` is typically constructed using
+    :func:`interval_range`:
+
+    >>> pd.interval_range(start=0, end=5)
+    IntervalIndex([(0, 1], (1, 2], (2, 3], (3, 4], (4, 5]]
+                  closed='right', dtype='interval[int64]')
+
+    It may also be constructed using one of the constructor
+    methods :meth:`IntervalIndex.from_arrays``,
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/interval.py]: double -> single closing backtick

File: pandas/core/indexes/interval.py [hunk 2]
```diff
@@ -120,9 +120,42 @@ class IntervalIndex(IntervalMixin, Index):
...
+    :meth:`IntervalIndex.from_breaks``, :meth:`IntervalIndex.from_intervals`
+    and :meth:`IntervalIndex.from_tuples``.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/interval.py]: and the above two lines as well

File: pandas/core/indexes/interval.py [hunk 3]
```diff
@@ -415,7 +448,9 @@ def from_tuples(cls, data, closed='right', name=None, copy=False):
 
         Examples
         --------
-
+        >>>  pd.IntervalIndex.from_tuples([(0, 1), (1,2)])
+        IntervalIndex([(0, 1], (1, 2]],
+                      closed='right', dtype='interval[int64]')
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/interval.py]: We could add here 'See also' to the other `from_` constructors

File: pandas/core/indexes/interval.py [hunk 4]
```diff
@@ -120,9 +120,42 @@ class IntervalIndex(IntervalMixin, Index):
...
+    See the `user guide
+    <http://pandas.pydata.org/pandas-docs/stable/advanced.html#intervalindex>`_
+    for more.
+
     See Also
     --------
     Index
+    Interval : A bounded slice-like interval
+    interval_range : Function to create a fixed frequency IntervalIndex
+    IntervalIndex.from_arrays : Construct an IntervalIndex from a a left and
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/interval.py]: 'a a'

File: pandas/core/indexes/interval.py [hunk 5]
```diff
@@ -120,9 +120,42 @@ class IntervalIndex(IntervalMixin, Index):
...
+                                right array
+    IntervalIndex.from_breaks : Construct an IntervalIndex from an array of
+                                splits
+    IntervalIndex.from_intervals : Construct an IntervalIndex from a array of
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/interval.py]: a -> an

**Review by jorisvandenbossche — COMMENTED** [review]
Cool!

**Comment by topper-123** [comment]
Ok, all comments taken care of.

**Comment by topper-123** [comment]
Hi @jorisvandenbossche . I've found, that the output string from the examples in the ``from_*`` methods is wrong, and I'm guessing they are from an older version of ``IntervalIndex``. The second commit proposal is regarding that.

In addition, the time-based examples for ``pd.interval_range`` don't work. I'l submit a issue regarding that as it must be a code issue.

**Comment by jorisvandenbossche** [comment]
> I'm guessing they are from an older version of IntervalIndex

Yes, the repr was changes during the PR process of adding the IntervalIndex, so not examples were updated for that

**Comment by jorisvandenbossche** [comment]
@topper-123 Thanks a lot!

**Pull request merged**